samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt

```diff
@@ -101,33 +107,34 @@ open class MainActivity : AppCompatActivity(), LoginFragment.OnLoginCompleteList
     }
 
     private fun displayAndPersistProfile(code: String, state: String) {
-        val handleAuth = { oauthInfo: OAuthInfo ->
-            val permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        launch {
+            val account = whenAccount.await()
+            val oauthInfo = account.completeOAuthFlow(code, state).await()
+
+            val permissionCheck = ContextCompat.checkSelfPermission(this@MainActivity,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE)
 
             if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                    requestPermissions(arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE), 1)
+                    requestPermissions(arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,
+                            Manifest.permission.WRITE_EXTERNAL_STORAGE), 1)
                 }
             }
 
-            val appFiles = this.applicationContext.getExternalFilesDir(null)
+            val appFiles = this@MainActivity.applicationContext.getExternalFilesDir(null)
             val logins = SyncLoginsClient(appFiles.absolutePath + "/logins.sqlite")
-            val tokenServer = account!!.getTokenServerEndpointURL()!!
+            val tokenServer = account.getTokenServerEndpointURL()
             logins.syncAndGetPasswords(oauthInfo, tokenServer).thenCatch { e ->
                 SyncResult.fromException(e)
             }.whenComplete { syncLogins ->
-                runOnUiThread {
-                    Toast.makeText(this, "Logins success", Toast.LENGTH_SHORT).show()
+                launch(UI) {
```

Comment:
This needs to stay `runOnUiThread` (for now). Otherwise, the block below is not being executed. We have an issue where we're tracking upgrading coroutines #1181 . Let's switch this back to `runOnUiThread` and address later? There's more here that needs to change.